### PR TITLE
v1.17.1 stage

### PR DESCRIPTION
**Pull request by adnxn**
### Summary

staging 1.17.1 release.

### Testing

- [x] Builds on Linux (`make release`)
- [x] Builds on Windows (`go build -out amazon-ecs-agent.exe ./agent`)
- [x] Unit tests on Linux (`make test`) pass
- [x] Unit tests on Windows (`go test -timeout=25s ./agent/...`) pass
- [x] Integration tests on Linux (`make run-integ-tests`) pass
- [x] Integration tests on Windows (`.\scripts\run-integ-tests.ps1`) pass
- [x] Functional tests on Linux (`make run-functional-tests`) pass
- [x] Functional tests on Windows (`.\scripts\run-functional-tests.ps1`) pass

New tests cover the changes: 

### Licensing

This contribution is under the terms of the Apache 2.0 License: yes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Changelog
 
-## 1.17.1-dev
+## 1.17.1
 * Bug - Fixed a bug that was causing a runtime panic by accessing negative index in the health check log slice. [#1239](https://github.com/aws/amazon-ecs-agent/pull/1239)
```
Comment (richardpen) [inline CHANGELOG.md]: Can you break this line to 80 characters and also get rid of the `.` at the end?
Also just want to confirm don't we need a changelog for #1240?

Comment (sharanyad) [reply CHANGELOG.md]: there is a changelog for https://github.com/aws/amazon-ecs-agent/pull/1240

Comment (adnxn) [reply CHANGELOG.md]: will fix char count and `.` at the end. For changelog entry, I think adding it to release notes should suffice. @aaithal, @sharanyad?

Comment (richardpen) [reply CHANGELOG.md]: Sorry, I missed the last line. The changelog is already there.

**Pull request merged**